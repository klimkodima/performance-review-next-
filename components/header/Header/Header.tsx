import { memo, ReactElement } from 'react';
import IconButton from '@mui/material/IconButton';
import { useTranslation } from 'next-i18next';

import styles from './header.module.css';

type HeaderPropsType = {
  onLogoutClick: () => void;
};

const Header = memo(({ onLogoutClick }: HeaderPropsType): ReactElement => {
  const { t } = useTranslation();
  return (
    <header className={styles.header}>
      <div className={styles.header_wrapper}>
        <h2 className={styles.dashboard_name}>{t('header.title')}</h2>
        <div className={styles.header_logoIcon}>
          <img src='/assets/icons/header-icon.png' alt='logo icon' />
        </div>
        <div>
          <IconButton className={styles.logoutIcon} onClick={onLogoutClick}>
            <img src='/assets/icons/logout-icon.png' alt='logout icon' />
          </IconButton>
        </div>
      </div>
    </header>
  );
});

export default Header;

Header.displayName = 'Header';
