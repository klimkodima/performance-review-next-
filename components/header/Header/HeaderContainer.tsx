import { ReactElement, useCallback } from 'react';
import Header from './Header';

const HeaderContainer = (): ReactElement => {
  const handleLogOutClick = useCallback(() => {
    // action
  }, []);

  return <Header onLogoutClick={handleLogOutClick} />;
};

export default HeaderContainer;
