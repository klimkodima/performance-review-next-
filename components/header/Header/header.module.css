.header {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  background-color: #f0f0f0;
}

.header_wrapper {
  display: flex;
  align-items: center;
  border: 2px solid #d7d7d7;
  height: 60px;
  padding: 0 30px;
}

.header_logoIcon {
  flex-grow: 1;
}

.header_logoIcon img {
  width: 46px;
}

.logoutIcon {
  height: 45px;
  width: 45px;
}

.logoutIcon img {
  width: 35px;
  opacity: 0.5;
}

.dashboard-name {
  margin-right: 25px;
  color: #787878;
  font-weight: 700;
}