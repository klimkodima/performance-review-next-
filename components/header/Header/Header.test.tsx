import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  test('Matches snapshot', () => {
    const tree = renderer
      .create(<Header onLogoutClick={() => true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
