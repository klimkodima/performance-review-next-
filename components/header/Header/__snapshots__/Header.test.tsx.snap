// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches snapshot 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper"
  >
    <h2
      className="dashboard-name"
    >
      title
    </h2>
    <div
      className="header__logoIcon"
    >
      <img
        alt="logo icon"
        src="header-icon.png"
      />
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium logoutIcon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="logout icon"
          src="logout-icon.png"
        />
      </button>
    </div>
  </div>
</header>
`;
